### PR TITLE
feat: automatic tracing

### PR DESCRIPTION
**Pull request by JamieMagee**
Replaces manual log statements which log methods and arguments, with automatic logging on log level trace.

### CODE REVIEW

**Comment by rarkins** [comment]
Interesting. No performance impact when logging at info or debug?

**Comment by JamieMagee** [comment]
Yeah, should have no impact unless you set debug level to trace.

**Comment by rarkins** [comment]
I was curious to test it out but only got this far:

```
FATAL: Fatal error: this.log is not a function
       "err": {
         "message": "this.log is not a function",
         "stack": "TypeError: this.log is not a function\n    at NJSTrace.inject (/Users/rhys/src/renovate/node_modules/njstrace/njsTrace.js:73:7)\n    at levels (/Users/rhys/src/renovate/lib/logger/index.ts:129:5)\n    at Object.parseConfigs (/Users/rhys/src/renovate/lib/config/index.ts:70:3)\n    at processTicksAndRejections (internal/process/task_queues.js:94:5)\n    at Object.start (/Users/rhys/src/renovate/lib/workers/global/index.ts:64:18)\n    at /Users/rhys/src/renovate/lib/renovate.ts:9:22"
       }
```

**Comment by JamieMagee** [comment]
I am seeing the same thing when debugging, but if you do `yarn build` and `node ./dist/renovate.js` it works. I'm looking into why 🤔

File: lib/datasource/docker/index.ts [hunk 1]
```diff
@@ -199,7 +199,6 @@ async function getManifestResponse(
   repository: string,
   tag: string
 ): Promise<GotResponse> {
-  logger.debug(`getManifestResponse(${registry}, ${repository}, ${tag})`);
```
Comment (viceice) [inline lib/datasource/docker/index.ts]: Are we sure to remove those `debug` logging?

File: lib/logger/index.ts [hunk 2]
```diff
@@ -137,6 +125,9 @@ export /* istanbul ignore next */ function addStream(
 
 export function levels(name: string, level: bunyan.LogLevel): void {
   bunyanLogger.levels(name, level);
+  if (level === 'trace') {
+    inject({ formatter: new RenovateFormatter(logger) });
```
Comment (viceice) [inline lib/logger/index.ts]: we should call this on early startup from `lib/renovate.ts`.

```ts
const njstrace = inject({ formatter: new RenovateFormatter(logger), enabled: false });
```
```suggestion
   njstrace.enabled = true;
```
https://www.npmjs.com/package/njstrace#enabled-boolean

and mock it globally in `test/globals`

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/datasource/docker/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/datasource/docker/index.ts]: I don't mind it. We were never consistent anyway.

**Pull request closed without merging**